### ISSUE TITLE
Classes that define indexing policies are incomplete/not available

### ISSUE THREAD

**Issue by nahk-ivanov** [Azure/azure-cosmos-dotnet-v3#1176]
**Describe the bug**
All classes that are supposed to define indexing type are marked as `internal`, thus public API does not allow to define hash/range indexes. See [Index](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Resource/Settings/Index.cs) base class, for example.

`IncludedPath` only exposes `Path` property while `Indexes` collection is marked `internal`.

**To Reproduce**
Get latest SDK version (3.6.0) and try to create a container with hash index on a property.

**Expected behavior**
`Indexes` property on `IncludedPath` should be public along with all index types, like `HashIndex` and `RangeIndex`.

**Actual behavior**
All index type classes are marked `internal` and cannot be used when creating containers.

**Environment summary**
SDK Version: 3.6.0
OS Version (e.g. Windows, Linux, MacOSX): Windows

**Additional context**
Maybe I'm missing some APIs? I looked at the [index management](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos.Samples/Usage/IndexManagement/Program.cs) sample, which claims to be
> 4. Using range indexes

but there is nothing that defines a range index. I assume it is default, so any `IncludedPath` will be a range index, but none the less...
It would also be nice to have indexing path as a constructor argument on `ExcludedPath` and `IncludedPath` - it is required anyway and having to specify a property name every time you initialize the class is not convenient.

**Comment by nahk-ivanov** [comment]
It seems like they were public up until f0307f9. Commit description is not very helpful, but it mentions fluent APIs, so I suppose one should make use of [ContainerBuilder](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Fluent/Settings/ContainerBuilder.cs). It doesn't have any public constructors, but it seems like `Database.DefineContainer` (through override in `DatabaseCore`) will get you an instance.

Looking at it - through spaghetti of delegates and bunch of nested concrete generic classes (like interfaces don't exist) it will only get you as deep as [PathDefinition](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Fluent/Settings/PathsDefinition.cs), which seems to only allow to set the path, but not the index type.

Searching for `HashIndex` within this repo mostly yields results in the tests project, with only one reference to `HashIndex` constructor in the SDK source code - static factory method `Hash` on the `Index` class itself. Ironically, there is an example in the xml-doc for that method:

```csharp
/// <example>
/// Here is an example to create HashIndex instance passing in the DataType:
/// <code language="c#">
/// <![CDATA[
/// HashIndex hashIndex = Index.Hash(DataType.String);
/// ]]>
/// </code>
/// </example>
```

Which of course you cannot do, because Index class is marked as `internal`.

So can we create non-default containers with this SDK?

**Comment by krishnan1985** [comment]
We have deprecated the IndexKind and Precision attribute from the database engine. All new containers now default to Range index with full precision for every path that is included, and this cannot be changed. The IndexKind and Precision attributes are ignored if they are sent (from prior versions of SDK, for example). IndexKind and precision can still be mutated on older containers which support this. This change accompanies a broader index layout improvement that we've rolled out for new containers, and will apply after slow migration for the rest of the containers in the near future.  

In line with this change, starting SDK version 3.0, we no longer allow setting the IndexKind or precision for paths.
